code/game/objects/items/melee/transforming.dm

```diff
@@ -87,3 +90,12 @@
 	if(clumsy_check && HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
 		to_chat(user, "<span class='warning'>You accidentally cut yourself with [src], like a doofus!</span>")
 		user.take_bodypart_damage(5,5)
+
+/obj/item/melee/transforming/proc/on_sharpen(datum/source, increment, max)
+	SIGNAL_HANDLER
+
+	if(sharpened_bonus)
+		return COMPONENT_BLOCK_SHARPEN_ALREADY
+	if(force_on + increment > max)
+		return COMPONENT_BLOCK_SHARPEN_MAXED
+	sharpened_bonus = increment
```

Comment:
how does this interact with the other bits of sharpening code that directly modify force, and what happens if it ought to be sharpened twice?